src/sentry/processing/realtime_metrics/base.py

```diff
@@ -41,6 +44,48 @@ def increment_project_event_counter(self, project_id: int, timestamp: float) ->
             pipeline.incr(key)
             pipeline.execute()
 
+    def get_lpq_candidates(self) -> Iterable[int]:
+        """
+        Returns IDs of all projects that should be considered for the low priority queue.
+        """
+
+        already_seen = set()
+        for item in self.inner.scan_iter(
+            match=f"{self._prefix}:*",
+        ):
+            _prefix, project_id_raw, _else = item.split(":")
+            project_id = _to_int(project_id_raw)
+            if project_id is not None and project_id not in already_seen:
+                already_seen.add(project_id)
+                yield project_id
+
+    def get_bucketed_counts_for_project(self, project_id: int) -> Iterable[BucketedCount]:
+        """
+        Returns a sorted list of timestamps (bucket size unknown) and the count of symbolicator
+        requests made during that timestamp for some given project.
+        """
+
+        # TODO: Should all entries be normalized against the current bucket size?
+        keys = sorted(
+            key
+            for key in self.inner.scan_iter(
+                match=f"{self._prefix}:{project_id}:*",
+            )
+        )
+        counts = self.inner.mget(keys)
+
+        for key, count_raw in zip(keys, counts):
+            _prefix, _project_id, timestamp_raw = key.split(":")
+
+            timestamp_bucket = _to_int(timestamp_raw)
+            count = _to_int(count_raw)
```

Comment:
so i've tried to make a best effort attempt to grab as much useful data as i can instead of erroring immediately, so that we could at least check projects that aren't malformed. this goes against what you suggested here, so i'd like to get your thoughts on this.

this does reach out to sentry if a malformed timestamp or count is found, for what it's worth.

Reply:
actually in retrospect i think it's fine to let the tasks and methods explode on a failure to `int()`. 

my original concern was that the side effect of doing so would be that a single malformed value in a redis store of otherwise well-formed values would break the entire automated system we have set up. what's changed my mind is the fact that we do have the manual kill switches, which means that we can manually assign or evict projects to the LPQ via those if the automated tasks are failing. the values of those kill switches also should take precedence over the automated kill switch's values, which means that we can fully control the list of projects in the LPQ exclusively through the switches.